controllers/watcher_controller.go

```diff
@@ -683,6 +691,35 @@ func (r *WatcherReconciler) ensureDBSync(
 	return ctrlResult, nil
 }
 
+func (r *WatcherReconciler) createTopLevelSecret(
```

Comment:
I find name createTopLevelSecret a bit confussing as this secret is not top level (understanding top level as the one on top of Watcher, the one providing all the passwords for all services in openstack control plane). I'd say this would be createSubLevelSecret ?